terraform/modules/rds_migration/ecs.tf

```diff
@@ -0,0 +1,128 @@
+#tfsec:ignore:aws-ecs-enable-container-insight:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+resource "aws_ecs_cluster" "this" {
+  #checkov:skip=CKV_AWS_65:TODO CLDC-2542 enable container insights if necessary for logging/monitoring
+  name = "${var.prefix}-db-migration"
+}
+
+resource "aws_ecs_task_definition" "ad_hoc_db_migration" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-ad-hoc-db-migration"
+  cpu                      = var.db_migration_task_cpu
+  execution_role_arn       = var.ecs_task_execution_role_arn
+  memory                   = var.db_migration_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = var.ecs_task_role_arn
+
+  container_definitions = jsonencode([
+    {
+      name      = "db-migration"
+      essential = true
+      image     = var.ecr_repository_url
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.this.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      secrets = [
+        { Name = "DATABASE_URL", valueFrom = var.database_connection_string_arn },
+        { Name = "CF_PASSWORD", valueFrom = aws_secretsmanager_secret.cloudfoundry_password.arn },
+        { Name = "CF_SERVICE", valueFrom = aws_secretsmanager_secret.cloudfoundry_service.arn },
+        { Name = "CF_SPACE", valueFrom = aws_secretsmanager_secret.cloudfoundry_space.arn },
+        { Name = "CF_USERNAME", valueFrom = aws_secretsmanager_secret.cloudfoundry_username.arn }
+      ]
+    }
+  ])
+
+  ephemeral_storage {
+    size_in_gib = var.ecs_task_ephemeral_storage
+  }
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+
+  lifecycle {
+    # The image will be updated by deployments - irritatingly we can't ignore changes just to the image

```

Comment:
Will the image be updated by deployments? That's tbc right.

Reply:
Rachael has confirmed we'll do this manually, so I've removed all these lifecycle blocks and comments about images / deployments 👍